dIEM_violin_pipeline.R

```diff
@@ -311,449 +316,176 @@ if (algorithm == 1){
+  # from table Expected, choose columns "Disease", "HMDB.code", "Metabolite" and "Isobaric_Compounds_Names"
+  Expected_red <- Expected[ ,c(5,14,13,18)]
+  Expected_red <- Expected_red[!duplicated(Expected_red[,c(1,2)]),]
+  names(Expected_red) <- gsub("HMDB.code", "HMDB_code",  names(Expected_red))  
+  names(Expected_red) <- gsub("Metabolite", "HMDB_name", names(Expected_red))
   
-
-  metab.list0[[index+1]] <- summed[c(1:20),c(1,2)]
-  stofgroup_files[index+1] <- "top20_hoogst"
-  metab.list0[[index+2]] <- summed[c(1:10),c(1,2)]
-  stofgroup_files[index+2] <- "top10_laagst"
-
   
-  make_plots <- function(metab.list,stoftest,pt,zscore_cutoff,xaxis_cutoff,ThisProbScore,ratios_cutoff, ptcount) {
-    stoftest <- as.character(stoftest)
-    if (ncol(metab.list) > 2) {
-      # metab list are the alarm values, so reduce the data frame to 2 columns and save metab.list4
-      metab.list4 <- metab.list
-      metab.list <- metab.list[c(1,2)]
-    }
-    # extract the top 20 highest and top 10 lowest scoring metabolites
-    if (startsWith(stoftest, "top") & ptcount > 1) {
-      if (endsWith(stoftest, "hoogst")){
-        topX <- unique(summed[pt]) %>% slice_max(unique(summed[pt]),n = 20)
-        topX <- inner_join(topX, summed[,c(1,2,(ptcount+1))], by = pt)
-      } else {
-        topX <- unique(summed[pt]) %>% slice_min(unique(summed[pt]),n = 10)
-        topX <- inner_join(topX, summed[,c(1,2,(ptcount+1))], by = pt)
-      }
-      count = 0
-      listrm <- c()
-      b <- 100000.00000
-      for (rows in c(1:nrow(topX))) {
-        a <- topX[rows,1]
-        if (a==b) { count = count + 1 }
-        else { count = 0 }
-        if (count >= 6) { listrm <- c(listrm, rows) }
-        b <- a 
-      }
-      if (length(listrm) > 0) { 
-        metab.list <- topX[-(listrm),-1]
-        isos <- other_isobaric(metab.list,"infile",FALSE)
-        metab.list <- rbind(metab.list, isos)
-      } else {
-        metab.list <- topX[-1]
-      }
-      # replace the metab.list dataframe with new values
-      #metab.list <- topX[-1]
+  # Find all text files in the given folder, which contain metabolite lists of which
+  # each file will be a page in the pdf with violin plots.
+  # Make a PDF file for each of the categories in metabolite_dirs
+  metabolite_dirs <- list.files(path=path_txtfiles, full.names=FALSE, recursive=FALSE)
+  for (metabolite_dir in metabolite_dirs) {
+    # create a directory for the output PDFs
+    pdf_dir <- paste(output_dir, metabolite_dir, sep="/")
+    dir.create(pdf_dir)
+    metabolite_files <- list.files(path=paste(path_txtfiles, metabolite_dir, sep="/"), pattern="*.txt", full.names=FALSE, recursive=FALSE)
+
+    # put all metabolites into one list
+    metab_list_all <- list()
+    index = 0
+    cat("making plots from the input files:")
+    # open the text files and add each to a list of dataframes (metab.list0)
+    for (infile in metabolite_files) {
+      index = index + 1
```

Comment:
If you need an index then I would use the `1:len(...)` pattern. Als take a look at `seq_along` which is an alternative o `1:len()` (looks better I think...).